### PR TITLE
Added method for dynamic font loading

### PR DESCRIPTION
**Pull request by krista-koivisto**
Added two new methods to the template:

1. `GoogleFonts.asMap()` which returns a map where the key is the font family and the value is the corresponding `GoogleFonts` method.
2. `GoogleFonts.getFont(String fontFamily, {...})` which looks up the method by font family name using `GoogleFonts.asMap()`, runs it and returns the resulting `TextStyle`. The {...} "parameter" here represents the standard parameters for font methods, I decided to omit them for clarity.

Also, added a unit test which tests that `GoogleFonts.getFont` returns the same `TextStyle` instance as the regular method call does.

These additions were made in response to #34.

### CODE REVIEW

**Comment by krista-koivisto** [comment]
@googlebot I signed it!

File: example/lib/main.dart [hunk 1]
```diff
@@ -47,7 +47,7 @@ class _MyHomePageState extends State<MyHomePage> {
           children: <Widget>[
             Text(
               'You have pushed the button this many times:',
-              style: GoogleFonts.oswald(textStyle: display1),
+              style: GoogleFonts.getFont("Oswald", textStyle: display1),
```
Comment (clocksmith) [inline example/lib/main.dart:50]: nit: use single quotes for strings

Comment (clocksmith) [reply example/lib/main.dart:50]: Actually, I don't think this should be changed here. We would prefer users to use `GoogleFonts.oswald` over `GoogleFonts.getFont('Oswald'` unless that had to use the string based API

File: test/generated_font_methods_test.dart [hunk 2]
```diff
@@ -659,4 +659,15 @@ main() {
     // Style is preserved.
     expect(textTheme.subtitle.fontStyle, equals(FontStyle.italic));
   });
+
+  //////////////////
+  // Method tests //
+  //////////////////
+
+  testWidgets('getFont returns the correct font', (tester) async {
+    final dynamicFont = GoogleFonts.getFont("Roboto Mono");
```
Comment (clocksmith) [inline test/generated_font_methods_test.dart:668]: nit: single quotes for string

File: generator/google_fonts.tmpl [hunk 3]
```diff
@@ -29,6 +29,94 @@ class GoogleFonts {
...
+    Paint background,
+    List<ui.Shadow> shadows,
+    List<ui.FontFeature> fontFeatures,
+    TextDecoration decoration,
+    Color decorationColor,
+    TextDecorationStyle decorationStyle,
+    double decorationThickness,
+  })> asMap() => const {
+    {{#method}}
+    "{{fontFamilyDisplay}}": GoogleFonts.{{methodName}},
```
Comment (clocksmith) [inline generator/google_fonts.tmpl:58]: Single quotes can also be used here

**Comment by krista-koivisto** [comment]
Thank you, @clocksmith!

Applied requested changes, as well as:

 * Changed example in `README.md` to use single quotes.
 * Removed redundant word in comment for `getFont`.

**Pull request merged**